tests/broker-engine/notifications.robot

```diff
@@ -0,0 +1,1303 @@
+
+    ${content}    Create List    SERVICE NOTIFICATION: John_Doe;host_1;service_1;CRITICAL;command_notif;
+    ${result}    Ctn Find In Log With Timeout    ${engineLog0}    ${start}    ${content}    60
+    Should Be True    ${result}    No notification has been sent concerning a critical service
+
+    ## Time to set the service to UP  hard
+
+    Ctn Process Service Result Hard    host_1    service_1    ${0}    The service_1 is OK
+
+    ${content}    Create List    SERVICE NOTIFICATION: John_Doe;host_1;service_1;RECOVERY (OK);command_notif;
+    ${result}    Ctn Find In Log With Timeout    ${engineLog0}    ${start}    ${content}    60
+    Should Be True    ${result}    The notification recovery is not sent
+
+    Ctn Stop Engine
+    Ctn Kindly Stop Broker
+
+not3
+    [Documentation]    This test case configures a single service and verifies that a non-OK notification is sent after the service exits downtime.
+    [Tags]    broker    engine    services    hosts    notification
+    Ctn Config Engine    ${1}    ${1}    ${1}
+    Ctn Config Notifications
+    Ctn Engine Config Set Value In Hosts    0    host_1    notifications_enabled    1
+    Ctn Engine Config Set Value In Hosts    0    host_1    notification_options    d,r
+    Ctn Engine Config Set Value In Hosts    0    host_1    contacts    John_Doe
+    Ctn Engine Config Set Value In Services    0    service_1    contacts    John_Doe
+    Ctn Engine Config Set Value In Services    0    service_1    notification_options    w,c,r
+    Ctn Engine Config Set Value In Services    0    service_1    notifications_enabled    1
+    Ctn Engine Config Set Value In Services    0    service_1    notification_period    24x7
+    Ctn Engine Config Replace Value In Services    0    service_1    check_interval    1
+    Ctn Engine Config Replace Value In Services    0    service_1    retry_interval    1
+    Ctn Engine Config Set Value In Contacts    0    John_Doe    host_notification_commands    command_notif
+    Ctn Engine Config Set Value In Contacts    0    John_Doe    service_notification_commands    command_notif
+
+    ${start}    Get Current Date
+    Ctn Start Broker
+    Ctn Start Engine
+
+    # Let's wait for the external command check start
+    ${content}    Create List    check_for_external_commands()
+    ${result}    Ctn Find In Log With Timeout    ${engineLog0}    ${start}    ${content}    60
+    Should Be True    ${result}    check_for_external_commands() should be available.
+
+    # It's time to schedule a downtime
+    Ctn Schedule Service Fixed Downtime    host_1    service_1    3600
+
+    ${result}    Ctn Check Service Downtime With Timeout    host_1    service_1    1    60
+    Should Be True    ${result}    service must be in downtime
+
+    Ctn Delete Service Downtime    host_1    service_1
+
+    Ctn Process Service Result Hard    host_1    service_1    ${2}    The service_1 is CRITICAL
+
```

Comment:
This test is not useful.
* You create a service
* You put a downtime on it.
* you remove the downtime
* you set the service state to CRITICAL HARD.
* And then you check a notification is well sent.

Why the downtime would have an impact on the service if it is removed?

An interesting test would be:
* Create the service
* Put a downtime on it.
* Set its state to CRITICAL HARD.
* Check no notification is sent
* Remove the downtime
* Check there is a notification
* Set the state to OK
* Check there is a recovery notification.